sdk/core/Azure.Core/src/RehydrationToken.Serialization.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+#nullable disable
+
+namespace Azure.Core
+{
+    public partial struct RehydrationToken : IJsonModel<RehydrationToken>
```

Comment:
In order to be used by the generic methods on MRW you need to implement IJsonModel<object> as well